operator/pkg/images/images.go

```diff
@@ -0,0 +1,235 @@
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	envCustomImageRegistry            = "CUSTOM_IMAGE_REGISTRY"
+	envCustomImageRegistryPullSecrets = "CUSTOM_IMAGE_REGISTRY_PULL_SECRETS"
+	EnvBaseImageOS                    = "BASE_IMAGE_OS"
+)
+
+// How to add new images:
+//
+// 1. Add a new Image enum value below
+// 2. Update imageLookupMap with image url
+// 3. If the image is a Cassandra/DSE image, also update maps:
+//    - versionToOSSCassandra
+//    - versionToDSE
+// 4. Additionally, if there a is a UBI Cassandra/DSE image
+//    available, also update:
+//    - versionToUBIOSSCassandra
+//    - versionToUBIDSE
+//
+
+type Image int
+
+// IMPORTANT: Only Image enum values (and ImageEnumLength) should go in  this
+// const expression
+const (
+	Cassandra_3_11_6 Image = iota
+	Cassandra_3_11_7
+	Cassandra_4_0_0
+
+	UBICassandra_3_11_6
+	UBICassandra_3_11_7
+	UBICassandra_4_0_0
+
+	DSE_6_8_0
+	DSE_6_8_1
+	DSE_6_8_2
+	DSE_6_8_3
+
+	UBIDSE_6_8_0
+	UBIDSE_6_8_1
+	UBIDSE_6_8_2
+	UBIDSE_6_8_3
+
+	ConfigBuilder
+	UBIConfigBuilder
+
+	BusyBox
+	Reaper
+	BaseImageOS
+
+	// NOTE: This line MUST be last in the const expression
+	ImageEnumLength int = iota
+)
+
+var imageLookupMap map[Image]string = map[Image]string {
+
+	Cassandra_3_11_6: "datastax/cassandra-mgmtapi-3_11_6:v0.1.5",
+	Cassandra_3_11_7: "datastax/cassandra-mgmtapi-3_11_7:v0.1.12",
+	Cassandra_4_0_0:  "datastax/cassandra-mgmtapi-4_0_0:v0.1.12",
+
+	UBICassandra_3_11_6: "datastax/cassandra:3.11.6-ubi7",
+	UBICassandra_3_11_7: "datastax/cassandra:3.11.7-ubi7",
+	UBICassandra_4_0_0:  "datastax/cassandra:4.0-ubi7",
+
+	DSE_6_8_0: "datastax/dse-server:6.8.0",
+	DSE_6_8_1: "datastax/dse-server:6.8.1",
+	DSE_6_8_2: "datastax/dse-server:6.8.2",
+	DSE_6_8_3: "datastax/dse-server:6.8.3",
+
+	UBIDSE_6_8_0: "datastax/dse-server:6.8.0-ubi7",
+	UBIDSE_6_8_1: "datastax/dse-server:6.8.1-ubi7",
+	UBIDSE_6_8_2: "datastax/dse-server:6.8.2-ubi7",
+	UBIDSE_6_8_3: "datastax/dse-server:6.8.3-ubi7",
+
+	ConfigBuilder:    "datastax/cass-config-builder:1.0.3",
+	UBIConfigBuilder: "datastax/cass-config-builder:1.0.3-ubi7",
+
+	BusyBox:     "busybox",
+	Reaper:      "thelastpickle/cassandra-reaper:2.0.5",
+}
+
+var versionToOSSCassandra map[string]Image = map[string]Image {
+	"3.11.6": Cassandra_3_11_6,
+	"3.11.7": Cassandra_3_11_7,
+	"4.0.0":  Cassandra_4_0_0,
+}
+
+var versionToUBIOSSCassandra map[string]Image = map[string]Image {
+	"3.11.6": UBICassandra_3_11_6,
+	"3.11.7": UBICassandra_3_11_7,
+	"4.0.0":  UBICassandra_4_0_0,
+}
+
+var versionToDSE map[string]Image = map[string]Image {
+	"6.8.0": DSE_6_8_0,
+	"6.8.1": DSE_6_8_1,
+	"6.8.2": DSE_6_8_2,
+	"6.8.3": DSE_6_8_3,
+}
+
+var versionToUBIDSE map[string]Image = map[string]Image {
+	"6.8.0": UBIDSE_6_8_0,
+	"6.8.1": UBIDSE_6_8_1,
+	"6.8.2": UBIDSE_6_8_2,
+	"6.8.3": UBIDSE_6_8_3,
+}
+
+var log = logf.Log.WithName("images")
+
+func getCustomImageRegistry() string {
+	return os.Getenv(envCustomImageRegistry)
```

Comment:
I'm wondering if the cluster-wide operator request plus the override registry request need to work better together, and if an env var won't work - we should ask @BrynCooke / VMWare

Reply:
For VMWare, the registry is given to us only when bringing up the operator (it's a setting in the template for the operator). I don't see how an environment variable could not work. If VMWare does a clusterwide install of the operator, then there will only be one such registry for the entire cluster.